clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -1054,6 +1079,73 @@ public AdminResponse getConsumptionSnapshots(String topicName, ComplementSet<Int
     return response;
   }
 
+  /**
+   * Return the metadata information for the given store. The data is retrieved from its respective repositories which
+   * originate from the VeniceServer.
+   * @param storeName
+   * @return {@link MetadataResponse} object that holds all the information required for answering a server metadata
+   * fetch request.
+   */
+  @Override
+  public MetadataResponse getMetadata(String storeName) {
+    MetadataResponse response = new MetadataResponse();
+    try {
+      Store store = metadataRepo.getStoreOrThrow(storeName);
+
+      Map<CharSequence, CharSequence> partitionerParams =
+          new HashMap<>(store.getPartitionerConfig().getPartitionerParams());
+      VersionProperties versionProperties = new VersionProperties(
+          store.getCurrentVersion(),
+          store.getVersion(store.getCurrentVersion()).get().getCompressionStrategy().getValue(),
+          store.getPartitionCount(),
+          store.getPartitionerConfig().getPartitionerClass(),
+          partitionerParams,
+          store.getPartitionerConfig().getAmplificationFactor());
+
+      Map<CharSequence, CharSequence> keySchema = Collections.singletonMap(
+          String.valueOf(schemaRepo.getKeySchema(storeName).getId()),
+          schemaRepo.getKeySchema(storeName).getSchema().toString());
+      Map<CharSequence, CharSequence> valueSchemas = new HashMap<>();
+      for (SchemaEntry schemaEntry: schemaRepo.getValueSchemas(storeName)) {
+        String valueSchemaStr = schemaEntry.getSchema().toString();
+        int valueSchemaId = schemaRepo.getValueSchemaId(storeName, valueSchemaStr);
+        valueSchemas.put(String.valueOf(valueSchemaId), valueSchemaStr);
+      }
+      int latestSuperSetValueSchemaId = store.getLatestSuperSetValueSchemaId();
+
+      Map<CharSequence, List<CharSequence>> routingInfo = new HashMap<>();
+      for (String resource: routingRepository.getResourceAssignment().getAssignedResources()) {
```

Comment:
Do we still need the `routingRepository`? I think we should be able to get everything we need with just the `customizedViewRepository` right?